mypy/checker.py

```diff
@@ -6152,13 +6147,12 @@ def handle_partial_var_type(
             return fixup_partial_type(typ)
 
     def is_defined_in_base_class(self, var: Var) -> bool:
-        if var.info:
-            for base in var.info.mro[1:]:
-                if base.get(var.name) is not None:
-                    return True
-            if var.info.fallback_to_any:
-                return True
-        return False
+        if not var.info:
+            return False
+        return (
+            any(base.get(var.name) is not None for base in var.info.mro[1:])
+            or var.info.fallback_to_any
```

Comment:
```suggestion
            var.info.fallback_to_any
            or any(base.get(var.name) is not None for base in var.info.mro[1:])
```

Reply:
This seems like a reasonable change, although unlike the other ones it isn't strictly equivalent.